### PR TITLE
Inform user if search returned 0 results

### PR DESCRIPTION
**Pull request by kaushalmodi**
This commit fixes the issue where the "Contacting host: .." messages stays
hanging in the echo area whether or not a query ended up with non-zero result
pods.

Now after all the pods are retrieved and displayed in the WolframAlpha buffer,
the echo area is cleared out.. so the "Contacting host: .." message doesn't stay
hanging.

If the query results in 0 retrieved pods, a "No results for your query." is
printed in the WolframAlpha buffer, so that the user doesn't have to keep on
waiting for something to show up.

* wolfram--delete-in-progress-notification: New function. The logic for searching backward for the "(searching)" notification and removing
  it is moved to this function.

* wolfram--append-pods-to-buffer: Use the above function, remove the unused
  `buffer` argument.

* wolfram--query-callback: New function. Moved out the lambda callback function
as this separate function for better readability and "debuggablity".

### CODE REVIEW

File: wolfram.el [hunk 1]
```diff
@@ -138,16 +138,24 @@
...
   (let ((inhibit-read-only t))
-    (goto-char (point-max))
-    (search-backward " (searching)")
-    (replace-match "")
-    (goto-char (point-max))
     (dolist (pod pods)
-      (wolfram--append-pod pod))
-    (insert "\n")))
+      (wolfram--append-pod pod)
+      (insert "\n"))))
```
Comment (hsjunnesson) [inline wolfram.el]: Please change this back. I insert a newline after all of the pods have been inserted. This makes sure all pods have one newline before the next heading.
With this change, there are two newlines before the next heading.

**Review by hsjunnesson — CHANGES_REQUESTED** [review]
Good job, thanks for these changes. However, I have one small aesthetic change.

File: wolfram.el (hunk 1 above)
Comment (kaushalmodi) [reply wolfram.el]: Nice catch! It was an overlook on my part. I did not intend the headings to be separated by 2 newlines. I have now fixed it.

**Comment by kaushalmodi** [comment]
@hsjunnesson Thanks for the review. I have force-pushed the fix for the extra newline on the same branch.

**Review by hsjunnesson — APPROVED** [review]

**Comment by hsjunnesson** [comment]
Looks great.

**Pull request merged**